performance/SwiftUIPerformanceQuickstart/PerformanceExample/Design/SYMBOLS.md

```diff
@@ -0,0 +1,371 @@
+    - [updateSaliencyMapAsync](#updatesaliencymapasync)
+    - [downloadImageAsync](#downloadimageasync)
+    - [classifyImageAsync](#classifyimageasync)
+    - [generateSaliencyMapAsync](#generatesaliencymapasync)
+    - [uploadSaliencyMapAsync](#uploadsaliencymapasync)
+    - [updateStatus](#updatestatus)
+    - [updateImage](#updateimage)
+    - [updateSaliencyMap](#updatesaliencymap)
+    - [downloadImage](#downloadimage)
+    - [classifyImage](#classifyimage)
+    - [generateSaliencyMap](#generatesaliencymap)
+    - [uploadSaliencyMap](#uploadsaliencymap)
+    - [makeTrace](#maketrace)
+- [ProcessDerivatives](#processderivatives)
+    - [ProcessTask](#processtask)
+    - [ProcessStatus](#processstatus)
+        - [text](#text)
+        - [view](#view)
+- [MainView](#mainview)
+    - [process](#process-1)
+    - [body](#body-1)
+- [DownloadView](#downloadview)
+    - [process](#process-2)
+    - [body](#body-2)
+- [ClassifyView](#classifyview)
+    - [process](#process-3)
+    - [body](#body-3)
+- [SaliencyMapView](#saliencymapview)
+    - [process](#process-4)
+    - [body](#body-4)
+- [UploadView](#uploadview)
+    - [process](#process-5)
+    - [body](#body-5)
+
+## PerformanceExampleApp
+```swift
+@main
+struct PerformanceExampleApp: App
+```
+main point of entry into app
+
+### init
+```swift
+init()
+```
+configures the FirebaseApp
+
+### body
+```swift
+var body: some Scene
+```
+returns a WindowGroup containing a MainView
+
+## Process
+```swift
+class Process: ObservableObject
+```
+handles requests for running different processes
+
+### status
+```swift
+@Published var status: ProcessStatus
+```
+publishes the status of the current process
+
+### image
+```swift
+@Published var image: UIImage?
+```
+stores the downloaded image
+
+### saliencyMap
+```swift
+@Published var saliencyMap: UIImage?
+```
+stores the generated saliency map
+
+### uploadSucceeded
+```swift
+var uploadSucceeded : Bool
+```
+indicates the status of the upload task
+
+### categories
+```swift
+var categories: [(category: String, confidence: VNConfidence)]?
+```
+stores the results of the classification task
+
+### context
+```swift
+lazy var context: CIContext { get set }
+```
+stores the CIContext used to generate the saliency map, initializing it when it needs to
+
+### isRunning
+```swift
+var isRunning: Bool { get }
+```
+returns whether a process is currently running
+
+### precision
+```swift
+let precision: Float
+```
+stores the precision used for the classification task
+
+### site
+```swift
+let site : String
+```
+stores the site from which to download the image
+
+### updateStatusAsync
+```swift
+@MainActor @available(iOS 15, tvOS 15,*)
+func updateStatusAsync(to newStatus: ProcessStatus, updateUploadStatus: Bool = false)
+```
+updates the Process status asynchronously, possibly updating the status of the upload task
+
+### updateImageAsync
+```swift
+@MainActor @available(iOS 15, tvOS 15, *) func updateImageAsync(to newImage: UIImage?)
```

Comment:
I think breaking these code snippets into two lines (one for function attributes, one for function signature) like done above would make them easier to read.